compiler/noirc_evaluator/src/ssa/plonky2_gen/console_asm_writer.rs

```diff
@@ -165,6 +170,23 @@ impl Display for U32TargetDisplay {
 pub(crate) struct ConsoleAsmWriter {
     pub builder: P2Builder,
     pub show_plonky2: bool,
+    file: Option<BufWriter<File>>,
+}
+
+impl ConsoleAsmWriter {
+    fn output_enabled(&self) -> bool {
+        self.show_plonky2 || self.file.is_some()
+    }
+
+    fn handle_output(&mut self, s: String) {
+        if self.show_plonky2 {
+            println!("{}", s);
+        }
+
+        if let Some(f) = &mut self.file {
+            writeln!(f, "{}", s).expect("Unable to write PLONKY2 data to file");
+        }
+    }
```

Comment:
I would argue that the struct name is not so good anymore, now that it also writes to a file... :-)

How about we simplify things by passing an instance of AsmWriter to Builder::new and we don't have it parametrized by type and we don't have a separate impl and trait?

Reply:
Ok. I removed the trait AsmWriter and renamed the struct ConsoleAndFileAsmWriter to AsmWriter.

Reply:
Great! Thanks!